packages/grafana-data/src/types/dataLink.ts

```diff
@@ -44,7 +44,7 @@ export interface DataLink<T extends DataQuery = any> {
 export interface InternalDataLink<T extends DataQuery = any> {
   query: T;
   datasourceUid: string;
-  datasourceName: string;
+  datasourceName: string; // used as a title if one is not defined
```

Comment:
**Suggestion:** I like this addition, although I would probably rephrase it a little bit to make it clearer  
```suggestion
  datasourceName: string; // used as a title if `DataLink.title` is empty
```

Reply:
@leventebalogh actually looking at this, I think we can remove it altogether. 

logic that uses datasourceName: https://github.com/grafana/grafana/blob/main/packages/grafana-data/src/utils/dataLinks.ts#L48